homeassistant/components/image/__init__.py

```diff
@@ -383,3 +408,37 @@ async def handle(
     ) -> web.StreamResponse:
         """Serve image stream."""
         return await async_get_still_stream(request, image_entity)
+
+
+async def async_handle_snapshot_service(
+    image: ImageEntity, service_call: ServiceCall
+) -> None:
+    """Handle snapshot services calls."""
+    hass = image.hass
+    filename: Template = service_call.data[ATTR_FILENAME]
+    filename.hass = hass
+
+    snapshot_file = filename.async_render(variables={ATTR_ENTITY_ID: image})
```

Comment:
The existing function uses the variable `entity_id` but passes the actual entity. I think this should `image.entity_id` to be more correct here. (I realize this is following what the existing service is doing, but I believe its not correct)